src/content/reference/react/captureOwnerStack.md

```diff
@@ -0,0 +1,164 @@
+
+For example, consider the following code:
+
+```tsx
+import * as React from 'react';
+import * as ReactDOMClient from 'react-dom/client';
+
+function SubComponent({disabled}) {
+  if (disabled) {
+    throw new Error('disabled');
+  }
+}
+
+function Component({label}) {
+  return (
+    <fieldset>
+      <legend>{label}</legend>
+      <SubComponent key={label} disabled={label === 'disabled'} />
+    </fieldset>
+  );
+}
+
+function Navigation() {
+  return null;
+}
+
+function App({children}) {
+  return (
+    <React.Suspense fallback="loading...">
+      <main>
+        <Navigation />
+        {children}
+      </main>
+    </React.Suspense>
+  );
+}
+
+createRoot(document.createElement('div')).render(
+  <App>
+    <Component label="disabled" />
+  </App>
+);
+```
+
+`SubComponent` would throw an error.
+The **parent** component stack of that error would be
+
+```
+at SubComponent
+at fieldset
+at Component
+at main
+at React.Suspense
+at App
+```
+
+However, the owner stack would only read
+
+```
+at SubComponent
+at Component
+```
+
+Neither `App` nor the DOM components (e.g. `fieldset`) are considered owners in this stack since they didn't contribute to "creating" the node containing `SubComponent`. `App` and DOM components only forwarded the node. `App` just rendered the `children` node as opposed to `Component` which created a node containing `SubComponent` via `<SubComponent />`.
+
+Neither `Navigation` nor `legend` are in the stack at all since it's only a sibling to a node containing `<SubComponent />`.
+
+### When to use which {/*when-to-use-which*/}
+
+The parent stack is useful for contextual information e.g. [`React.Suspense`](/reference/react/Suspense), [React Context](https://react.dev/reference/react/createContext), or [`<form>`](/reference/react-dom/components/form).
+
+The owner stack is useful for tracing the flow of props. Owner stacks are equivalent to [call stacks](https://developer.mozilla.org/en-US/docs/Glossary/Call_stack) if all JSX would be converted to direct function calls e.g. `Component({label: "disabled"})` instead of `<Component label="disabled" />`.
+
+## Usage {/*usage*/}
+
+### Expanding error stacks {/*expanding-error-stacks*/}
+
+In addition to the stack trace of the <CodeStep step={1}>error</CodeStep> itself, you can use <CodeStep step={2}>`captureOwnerStack`</CodeStep> to append the stack trace of the owner Component.
+This can help trace the error especially when the error is caused by props. The owner Component stack helps trace the flow of props.
+
+```jsx [[9, 15, "error"], [34, 10, "captureOwnerStack"]]
+import { captureOwnerStack } from 'react'
+import { hydrateRoot } from 'react-dom/client';
+
+const root = hydrateRoot(
+  document.getElementById('root'),
+  <App />,
+  {
+    onCaughtError: (error, errorInfo) => {
+      if (process.env.NODE_ENV !== 'production') {
+        const ownerStack = captureOwnerStack();
+        error.stack += ownerStack;
```

Comment:
Do we really recommend appending it to the existing stack? Won't that add it to the end?

IMO this should be like:

```
if (__DEV__) { 
  showDialog(error, ownerStack, componentStack);
} else {
  logProdError(error, componentStack);
}
```

Reply:
We do that in Next.js and it's conceptually how you arrive at the error if every JSX callsite would just be a function call.